### DISCUSSION TITLE
How to set ChannelCredentials in header?

### DISCUSSION THREAD

**Discussion by ncsibra** [AnWeber/httpyac#553] [General]
Hi,

I have a GRPC server which running in insecure mod in local but in dev env using secure connection.
Setting up this work with dev and removing work with local:

```js
{{@request
const grpc = require('@grpc/grpc-js');
request.channelCredentials = grpc.ChannelCredentials.createSsl();
}}
```

How can I make this environment dependent?

The documentation says: 

> Header ChannelCredentials or Authorization are special and defines the [authentication](https://grpc.io/docs/guides/auth/#nodejs) used by gRPC. If no such header is specified, grpc.credentials.createInsecure() is used automatically

I'm unable to figure out how to set the same thing through a header.
My current ugly solution is to set a variable in the environment files like `ssl=true` and use this code:

```js
{{@request
if (`${ssl}` == 'true') {
  const grpc = require('@grpc/grpc-js');
  request.channelCredentials = grpc.ChannelCredentials.createSsl();
}
}}
```

Is there a better way to do this?

**Comment by AnWeber** [comment]
@ncsibra Currently, there is no better way. There is already the possibility to fill a header with key `channelCredentials` directly, but this does not help you in your setup right now using. I add that you only have to specify a string ssl or insecure and then the ChannelCredentials object is created automatically. Then you should be able to simplify your use case.

```
GRPC grpcb.in:9001/hello.HelloService/sayHello
channelCredentials: {{channelCredentials}}
```

> insecure.env
```env
channelCredentials=insecure
```

> secure.env
```env
channelCredentials=ssl
```

This will work with next release. If you have an even better idea, feel free to share it.

**Reply by ncsibra** [reply]
Thanks for the answer, the solution sounds good to me for my current setup.
Just started to work with GRPC, so don't really know what else should be considered for channel credentials.

**Reply by AnWeber** [reply]
The same with me. I built the integration to simply have a use case to try out. But I don't have it in productive use.

**Reply by ncsibra** [reply]
Anyway, thanks for it, this is the best alternative I found after Postman deprecated scratchpad so keep up the good work. :)

**Reply by AnWeber** [reply]
thx.
